### DISCUSSION TITLE
How to do an analysis with additional masses

### DISCUSSION THREAD

**Discussion by hrntsm** [KratosMultiphysics/Kratos#10829] [Q&A] [answered]
I want to perform an Eigenanalysis on a model with shell elements and additional mass points.
I used NodalConcentratedElement3D1N to set up a mass point and run an eigenvalue analysis, but the eigenvalue results did not change with or without this element.

When I check NODAL_MASS in Paraview, it is set to "nan".
<img width="300" alt="image" src="https://user-images.githubusercontent.com/23289252/223037742-56cb1c32-d3ce-420e-9e47-d02b36303bc5.png">

I am assigning settings to the NODAL_MASS as properties in material_plate.json, but I think this is where I am going wrong.
```json
{
  "properties": [
    {
      "model_part_name": "plate.nodal_mass",
      "properties_id": 1,
      "Material": {
        "name": "mat_mass",
        "Variables": {
          "NODAL_MASS": 100
        }
      }
    }
  ]
}
```

If anyone has any solutions, I would appreciate an answer.
Thanks in advance.

Attached are the files used for analysis
[analysis_parameters_problem_eigen.zip](https://github.com/KratosMultiphysics/Kratos/files/10895055/analysis_parameters_problem_eigen.zip)

**Comment by hrntsm** [comment, marked as answer]
I found that the mass is calculated by setting it in the mdpa file as follows, not in the json that sets the Material.

```
Begin Elements NodalConcentratedElement3D1N
      100          0        1
End Elements

Begin ElementalData NODAL_MASS
      100  20000
End ElementalData
```

**Reply by rubenzorrilla** [reply]
I think that you can alternatively use the `AssignScalarVariableToElementsProcess` to do the same.

**Reply by hrntsm** [reply]
Thanks @rubenzorrilla.
I was able to set it up the following way you taught me.

```json
"process_list" : [
      {
        "python_module": "assign_scalar_variable_to_elements_process",
        "kratos_module": "KratosMultiphysics",
        "process_name": "AssignScalarVariableToElementsProcess",
        "Parameters": {
          "model_part_name": "plate.nodal_mass",
          "variable_name": "NODAL_MASS",
          "value": 10.0
        }
      }
]
```
